iroh/src/docs_engine.rs

```diff
@@ -46,20 +48,77 @@ pub struct Engine {
     actor_handle: SharedAbortingJoinHandle<()>,
     #[debug("ContentStatusCallback")]
     content_status_cb: ContentStatusCallback,
+    default_author: AuthorId,
+    default_author_storage: Arc<DefaultAuthorStorage>,
+}
+
+/// Where to persist the default author.
+///
+/// If set to `Mem`, a new author will be created in the docs store before spawning the sync
+/// engine. Changing the default author will not be persisted.
+///
+/// If set to `Persistent`, the default author will be loaded from and persisted to the specified
+/// path (as base32 encoded string of the author's public key).
+#[derive(Debug)]
+pub enum DefaultAuthorStorage {
+    Mem,
+    Persistent(PathBuf),
+}
+
+impl DefaultAuthorStorage {
+    pub async fn load(&self, docs_store: &SyncHandle) -> anyhow::Result<AuthorId> {
+        match self {
+            Self::Mem => {
+                let author = Author::new(&mut rand::thread_rng());
+                let author_id = author.id();
+                docs_store.import_author(author).await?;
+                Ok(author_id)
+            }
+            Self::Persistent(ref path) => {
+                if path.exists() {
+                    let data = tokio::fs::read_to_string(path).await?;
+                    let author_id = AuthorId::from_str(&data)?;
+                    if docs_store.export_author(author_id).await?.is_none() {
+                        bail!("The default author is missing from the docs store. To recover, delete the file `{}`. Then iroh will create a new default author.", path.to_string_lossy())
+                    }
+                    Ok(author_id)
+                } else {
+                    let author = Author::new(&mut rand::thread_rng());
+                    let author_id = author.id();
+                    docs_store.import_author(author).await?;
+                    tokio::fs::write(path, author_id.to_string()).await?;
+                    Ok(author_id)
+                }
+            }
+        }
+    }
+    pub async fn save(&self, docs_store: &SyncHandle, author_id: AuthorId) -> anyhow::Result<()> {
+        if docs_store.export_author(author_id).await?.is_none() {
+            bail!("The author does not exist");
+        }
+        match self {
+            Self::Mem => {}
```

Comment:
this is surprising, why not store it in memory?

Reply:
It is stored on the `DocsEngine` struct already, so `save` is a noop for mem storage. Will add a comment.

Reply:
OK pushed a new commit that makes this nicer (and actually work).